File: windows/src/GoalMaker.Core/Planning/TodaySections.cs
namespace GoalMaker.Core.Planning;

/// <summary>Today's open tasks by section (docs/lists.md), each in its order.</summary>
public sealed record TodaySections(
    IReadOnlyList<TaskItem> Priorities,
    IReadOnlyList<TaskItem> Scheduled,
    IReadOnlyList<TaskItem> More,
    IReadOnlyList<TaskItem> Overdue)
{
    public bool IsEmpty => Priorities.Count == 0 && Scheduled.Count == 0 && More.Count == 0 && Overdue.Count == 0;
}

File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/TodaySections.kt
package com.goalmaker.app.application.planning

/** Today's open tasks by section (docs/lists.md), each in its order. */
data class TodaySections(
    val priorities: List<TaskItem>,
    val scheduled: List<TaskItem>,
    val more: List<TaskItem>,
    val overdue: List<TaskItem>,
) {
    val isEmpty: Boolean get() = priorities.isEmpty() && scheduled.isEmpty() && more.isEmpty() && overdue.isEmpty()
}
